### ISSUE TITLE
html2text does a bad job of text extraction

### ISSUE THREAD

**Issue by bhaddow** [paracrawl/Malign#4]
Look at the output of `commoncrawl/download_candidates.py`, the html->text conversion can be quite bad in places. There are random new lines inserted in text passages, which means sentence alignment works poorly, and good parallel sentences could get thrown away. Also inserting markup isn't really useful for us.

Look at the way html2text handles this site: http://usategieta.com/index-eu.php

An alternative could be to use the bitextor convertor (not sure how good that is) or use readability (https://github.com/EdinburghNLP/news-crawl/blob/master/extract-text-corpus/readability/readability.py). The latter removes navigational text etc., which may also be a bad idea.

I thought this was worth recording, but I'm not sure if we are going to pursue extraction from CommonCrawl.

**Comment by bhaddow** [comment]
OK, I did a comparison. This is for English-Basque, where I extract the first 1000 domains from CommonCrawl (there are about 20000 that fitted my size criteria).

With html2text I get 2000 lines, whereas with readability I get 3600. html2text has url markup scattered through it. readability does not. Other than that, it is hard to say what the best corpus is:

/home/bhaddow/experiments/paracrawl/basque/parallel.1000.readability
vs
/home/bhaddow/experiments/paracrawl/basque/parallel.1000.html2text

**Comment by bhaddow** [comment]
I think one of the main takeaways is that (as @kpu points out for tokenizers) the html->text converter should be easily swappable - and we should have some way of benchmarking alternatives.

**Comment by hieuhoang** [comment]
I agree. I'll take a look at it after the mt marathon if no one else has

**Comment by bhaddow** [comment]
I also found this: http://corpus.tools/wiki/Justext

**Comment by jowagner** [comment]
The docs for the python module `html2text` (Alireza Savanddo and Randy Syring) do not mention whether it is based on the command line tool `html2text` (Arno Unkrig and Martin Bayer). It may be worthwhile testing the later as well.
